website/registrations/models.py

```diff
@@ -111,7 +128,13 @@ def save(
         if self.membership_type == Membership.BENEFACTOR:
             self.length = self.MEMBERSHIP_YEAR
         else:
-            self.contribution = settings.MEMBERSHIP_PRICES[self.length]
+            if self.membership_upgrade_discount_applies:
+                self.contribution = (
+                    settings.MEMBERSHIP_PRICES["study"]
```

Comment:
I think that these keys for the lengths are constants somewhere

Reply:
I checked and we use the strings everywhere in the code base (as these are settings, not values in the database)

Reply:
Will address in issue